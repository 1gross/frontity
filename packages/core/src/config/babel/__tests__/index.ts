import getBabel from "../";

test("Babel returns for development", () => {
  expect(getBabel("development")).toMatchSnapshot();
});

test("Babel returns for production", () => {
  expect(getBabel("production")).toMatchSnapshot();
});
