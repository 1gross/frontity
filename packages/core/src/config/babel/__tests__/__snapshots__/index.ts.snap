// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel returns for development 1`] = `
Object {
  "es5": Object {
    "plugins": Array [
      "@babel/plugin-syntax-dynamic-import",
      "@loadable/babel-plugin",
      "@babel/plugin-proposal-object-rest-spread",
      "@babel/plugin-proposal-class-properties",
      "babel-plugin-lodash",
    ],
    "presets": Array [
      "@babel/preset-typescript",
      Array [
        "@babel/preset-env",
        Object {
          "targets": Object {
            "browsers": Array [
              "and_chr >= 67",
              "and_ff >= 18",
              "and_uc >= 11.8",
              "android >= 67",
              "not android <= 4.4.4",
              "chrome >= 49",
              "edge >= 12",
              "firefox >= 18",
              "ios_saf >= 10",
              "not op_mini all",
              "op_mob >= 46",
              "opera >= 36",
              "safari >= 10",
              "samsung >= 5",
            ],
          },
          "useBuiltIns": "entry",
        },
      ],
      "@babel/preset-react",
    ],
  },
  "module": Object {
    "plugins": Array [
      "@babel/plugin-syntax-dynamic-import",
      "@loadable/babel-plugin",
      "@babel/plugin-proposal-object-rest-spread",
      "@babel/plugin-proposal-class-properties",
      "babel-plugin-lodash",
    ],
    "presets": Array [
      "@babel/preset-typescript",
      Array [
        "@babel/preset-env",
        Object {
          "targets": Object {
            "esmodules": true,
          },
          "useBuiltIns": false,
        },
      ],
      "@babel/preset-react",
    ],
  },
  "server": Object {
    "plugins": Array [
      "@babel/plugin-syntax-dynamic-import",
      "@loadable/babel-plugin",
      "@babel/plugin-proposal-object-rest-spread",
      "@babel/plugin-proposal-class-properties",
      "babel-plugin-lodash",
    ],
    "presets": Array [
      "@babel/preset-typescript",
      Array [
        "@babel/preset-env",
        Object {
          "targets": Object {
            "node": "8.10",
          },
          "useBuiltIns": false,
        },
      ],
      "@babel/preset-react",
    ],
  },
}
`;

exports[`Babel returns for production 1`] = `
Object {
  "es5": Object {
    "plugins": Array [
      "@babel/plugin-syntax-dynamic-import",
      "@loadable/babel-plugin",
      "@babel/plugin-proposal-object-rest-spread",
      "@babel/plugin-proposal-class-properties",
      "babel-plugin-lodash",
    ],
    "presets": Array [
      "@babel/preset-typescript",
      Array [
        "@babel/preset-env",
        Object {
          "targets": Object {
            "browsers": Array [
              "and_chr >= 67",
              "and_ff >= 18",
              "and_uc >= 11.8",
              "android >= 67",
              "not android <= 4.4.4",
              "chrome >= 49",
              "edge >= 12",
              "firefox >= 18",
              "ios_saf >= 10",
              "not op_mini all",
              "op_mob >= 46",
              "opera >= 36",
              "safari >= 10",
              "samsung >= 5",
            ],
          },
          "useBuiltIns": "entry",
        },
      ],
      "@babel/preset-react",
    ],
  },
  "module": Object {
    "plugins": Array [
      "@babel/plugin-syntax-dynamic-import",
      "@loadable/babel-plugin",
      "@babel/plugin-proposal-object-rest-spread",
      "@babel/plugin-proposal-class-properties",
      "babel-plugin-lodash",
    ],
    "presets": Array [
      "@babel/preset-typescript",
      Array [
        "@babel/preset-env",
        Object {
          "targets": Object {
            "esmodules": true,
          },
          "useBuiltIns": false,
        },
      ],
      "@babel/preset-react",
    ],
  },
  "server": Object {
    "plugins": Array [
      "@babel/plugin-syntax-dynamic-import",
      "@loadable/babel-plugin",
      "@babel/plugin-proposal-object-rest-spread",
      "@babel/plugin-proposal-class-properties",
      "babel-plugin-lodash",
    ],
    "presets": Array [
      "@babel/preset-typescript",
      Array [
        "@babel/preset-env",
        Object {
          "targets": Object {
            "node": "8.10",
          },
          "useBuiltIns": false,
        },
      ],
      "@babel/preset-react",
    ],
  },
}
`;
